framework/bootstrap/configcontroller.go

```diff
@@ -357,38 +352,56 @@ func (c *configController) RegisterEventHandler(kind resource.GroupVersionKind,
 	}
 }
 
-func (c *configController) Run(stop <-chan struct{}) {
+func (c *Controller) Run(stop <-chan struct{}) {
 	for i := range c.monitorControllers {
 		go c.monitorControllers[i].Run(stop)
 	}
 	<-stop
-	// stop k8s informers
-	c.stop <- struct{}{}
+	log.Infof("stop controller run")
+}
+
+func (c *Controller) Ready() bool {
+	for _, controller := range c.monitorControllers {
+		if !controller.Ready() {
+			log.Infof("config %+v not ready", controller.kind)
```

Comment:
`Ready` just tell the caller if the `Controller` is ready. In my opinion, which `monitorController` is not ready is more suitable to be logged as debug information.